tokio/src/io/async_fd.rs

```diff
@@ -243,6 +243,25 @@ impl<T: AsRawFd> AsyncFd<T> {
         Self::new_with_handle_and_interest(inner, scheduler::Handle::current(), interest)
     }
 
+    /// Create a new AsyncFd with the provided raw epoll flags for registration.
+    ///
+    /// These flags replace any epoll flags would normally set when registering the fd.
+    ///
+    ///  **Note**: This is an [unstable API][unstable]. The public API of this may break in 1.x
+    /// releases.
+    /// See [the documentation on unstable features][unstable] for details.
+    ///
+    ///  [unstable]: crate#unstable-features
+    #[track_caller]
+    #[cfg(all(target_os = "linux", tokio_unstable))]
+    #[cfg_attr(docsrs, doc(cfg(tokio_unstable)), doc(cfg(target_os = "linux")))]
+    pub fn with_epoll_flags(inner: T, flags: u32) -> io::Result<Self>
```

Comment:
Shouldn't this also take an interest? Otherwise we will eventually be asked to add an `with_epoll_flags_and_interest`.

Reply:
The flags are used to specify the interest here.

Because epoll shoves everything into one int, including interest and flags, and because there can be complicated rules around what flags and interests are allowable together, it's much more practical to tell users to specify the interest manually in the flags.

We'd only end up with confusing behavior if we chose to add both as fields. For example, some of the fields we specify for readability are mutually exclusive with EPOLLEXCLUSIVE, which would confuse users since they would probably try and use EPOLLEXCLUSIVE primarily with read interest on listeners.